### DISCUSSION TITLE
Hi, impressed by your work.

### DISCUSSION THREAD

**Discussion by juno-junho** [gautamkrishnar/blog-post-workflow#173] [Q&A]

**Comment by juno-junho** [comment]
Hi, thank you for providing amazing features in github.
It made my README page better.
I'm learning programming and I'm really curious about how does it work.
By just adding yml file in specific directory, my README.md files are updating consistently.
it's truely amaing, and Can you breifly tell me how it works?
Thank you

**Reply by gautamkrishnar** [reply]
Hi Juno, thanks for the kind words. GitHub automatically triggers the workflow periodically.  it works by executing the javascript code that is at https://github.com/gautamkrishnar/blog-post-workflow/blob/master/src/blog-post-workflow.js, which will in turn fetch the list of blog posts from your site using RSS. RSS is a global standard that websites use to generate machine parseable content from the blog posts you had written. Since it is a standard, the code can easily identify and list the blog posts written by you on your website. It then applies some [GitHub markdown](https://github.github.com/gfm/) magic to generate the list of blog posts in your readme.  Finally, it commits the generated changes to your repository.

**Reply by juno-junho** [reply]
Thanks a lot! I was wondering if I could ask one more question.
Do I need to know well how  workflow in github works, github markdown and javascipt?

**Reply by gautamkrishnar** [reply]
Workflow is created by following the GitHub actions developer docs docs: https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action
